pkg/armhelpers/mockclients.go

```diff
@@ -371,6 +462,57 @@ func (mc *MockAKSEngineClient) ListVirtualMachines(ctx context.Context, resource
 		}, errors.New("ListVirtualMachines failed")
 	}
 
+	if mc.FakeListVirtualMachineResult == nil {
+		mc.FakeListVirtualMachineResult = func() []compute.VirtualMachine {
+			return []compute.VirtualMachine{mc.MakeFakeVirtualMachine(DefaultFakeVMName, defaultK8sVersionForFakeVMs)}
+		}
+	}
+	vms := mc.FakeListVirtualMachineResult()
+	vmr := compute.VirtualMachineListResult{}
+	vmr.Value = &vms
+
+	return &MockVirtualMachineListResultPage{
+		Fn: func(lastResults compute.VirtualMachineListResult) (compute.VirtualMachineListResult, error) {
+			return compute.VirtualMachineListResult{}, nil
+		},
+		Vmlr: vmr,
+	}, nil
+}
+
+//ListVirtualMachineScaleSets mock
+func (mc *MockAKSEngineClient) ListVirtualMachineScaleSets(ctx context.Context, resourceGroup string) (VirtualMachineScaleSetListResultPage, error) {
+	if mc.FailListVirtualMachineScaleSets {
+		return &MockVirtualMachineScaleSetListResultPage{}, errors.New("ListVirtualMachines failed")
+	}
+	if mc.FakeListVirtualMachineScaleSetsResult == nil {
+		//return 0 machined by default
+		mc.FakeListVirtualMachineScaleSetsResult = func() []compute.VirtualMachineScaleSet {
+			return []compute.VirtualMachineScaleSet{}
+		}
+	}
+
+	vmsslr := compute.VirtualMachineScaleSetListResult{}
+	vmss := mc.FakeListVirtualMachineScaleSetsResult()
+	vmsslr.Value = &vmss
+
+	return &MockVirtualMachineScaleSetListResultPage{
+		Fn: func(compute.VirtualMachineScaleSetListResult) (compute.VirtualMachineScaleSetListResult, error) {
+			return compute.VirtualMachineScaleSetListResult{}, nil
+		},
+		Vmsslr: vmsslr,
+	}, nil
+}
+
+//GetVirtualMachine mock
+func (mc *MockAKSEngineClient) GetVirtualMachine(ctx context.Context, resourceGroup, name string) (compute.VirtualMachine, error) {
+	if mc.FailGetVirtualMachine {
+		return compute.VirtualMachine{}, errors.New("GetVirtualMachine failed")
+	}
+	return mc.MakeFakeVirtualMachine(DefaultFakeVMName, defaultK8sVersionForFakeVMs), nil
+}
+
+// MakeFakeVirtualMachineScaleSetVM creates a fake vmss vm
+func (mc *MockAKSEngineClient) MakeFakeVirtualMachineScaleSetVM(orchestratorTag string) compute.VirtualMachineScaleSetVM {
```

Comment:
`MockAKSEngineClient` already communicates that it is going to be a Fake VM 

```suggestion
func (mc *MockAKSEngineClient) CreateVirtualMachineScaleSetVM(orchestratorTag string) compute.VirtualMachineScaleSetVM {
```

Reply:
I think that looks like it's an actual method of what we are faking, versus a helper to provide you with an instance of a faked VM, so I'll keep it as is